foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/helper/ConversionManager.java

```diff
@@ -169,8 +175,10 @@ public Object clone() {
                 return null;
             }
         }
-
-        if ((sourceObject.getClass() == javaClass) || (javaClass == null) || (javaClass == ClassConstants.OBJECT) || (javaClass == ClassConstants.BLOB) || (javaClass == ClassConstants.CLOB)) {
+        if (sourceObject.getClass() == javaClass || javaClass == null || javaClass == ClassConstants.OBJECT
+                || javaClass == ClassConstants.BLOB || javaClass == ClassConstants.CLOB
+                // JSON has its own default converter registered
+                || javaClass == JsonValue.class || javaClass == JsonObject.class || javaClass == JsonArray.class) {
```

Comment:
I think, that `JsonValue, JsonObject, JsonArray` should be described in `ClassConstants` to be consistent with other types.

Reply:
I already did it before, but Lukas asked me to remove them from `ClassConstants`.